src/Concordium/Client/Runner.hs

```diff
@@ -1978,34 +2037,40 @@ processContractCmd action baseCfgDir verbose backend =
                           , ccParameter = wasmParameter
                           , ccEnergy = nrg
                           }
-      invokeContextArg <- case Text.decodeUtf8' . BSL.toStrict . AE.encode $ invokeContext of
-            Left _ -> logFatal ["Could not invoke contract due to internal error: decoding UTF-8 failed"] -- Should never happen.
-            Right text -> return text
 
-      withClient backend . withBestBlockHash block $ \bb -> do
-        res <- invokeContract invokeContextArg bb
-        case res of
-          Left err -> logFatal [[i|Invocation failed with error: #{err}|]]
-          Right (GRPCResponse _ jsonValue) -> case AE.fromJSON jsonValue of
-            AE.Error jsonErr -> logFatal [[i|Invocation failed with error: #{jsonErr}|]]
-            AE.Success InvokeContract.Failure{..} -> do
-              returnValueMsg <- mkReturnValueMsg rcrReturnValue schemaFile modSchema contractName updatedReceiveName True
-              -- Logs in cyan to indicate that the invocation returned with a failure.
-              -- This might be what you expected from the contract, so logWarn or logFatal should not be used.
-              log Info (Just ANSI.Cyan) [[iii|Invocation resulted in failure:\n
-                                                - Energy used: #{showNrg rcrUsedEnergy}\n
-                                                - Reason: #{showRejectReason verbose rcrReason}
-                                                #{returnValueMsg}|]]
-            AE.Success InvokeContract.Success{..} -> do
-              let eventsMsg = case mapMaybe (fmap (("  - " <>) . Text.pack) . showEvent verbose Nothing) rcrEvents of
-                                [] -> Text.empty
-                                evts -> [i|- Events:\n#{Text.intercalate "\n" evts}|]
-              returnValueMsg <- mkReturnValueMsg rcrReturnValue schemaFile modSchema contractName updatedReceiveName False
-              logSuccess [[iii|Invocation resulted in success:\n
-                               - Energy used: #{showNrg rcrUsedEnergy}
-                               #{returnValueMsg}
-                               #{eventsMsg}
-                               |]]
+      -- VH/FIXME: Should the blockhash input in the following be `Best` or `Given bbHash`?
```

Comment:
Yes, it should be the given block. That's the most consistent, even if it was different before.

Reply:
👍🏼